operator/v1/types_csi_cluster_driver.go

```diff
@@ -118,25 +121,116 @@ const (
+	// subscriptionID defines the Azure subscription that contains the disk encryption set.
+	// When omitted, the subscription from the authenticating credentials of the CSI driver
+	// will be used.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern:=^[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}$
+	SubscriptionID string `json:"subscriptionID"`
+
+	// resourceGroup defines the Azure resource group that contains the disk encryption set.
+	// When omitted, the cluster resource group from the cluster infrastructure object
+	// will be used.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=^[-\w\._\(\)]+$
+	ResourceGroup string `json:"resourceGroup"`
+
+	//name is the name of the disk encryption set that will be set on the default storage class.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength:=80
+	// +kubebuilder:validation:Pattern:=[a-zA-Z0-9_-]
+	Name string `json:"name"`
+}
+
+// AzureCSIDriverConfigSpec defines properties that can be configured for the Azure CSI driver.
+type AzureCSIDriverConfigSpec struct {
+	// diskEncryptionSet sets the cluster default storage class to encrypt volumes with a
+	// customer-managed encryption set, rather than the default platform-managed keys.
+	// +optional
+	DiskEncryptionSet *AzureDiskEncryptionSet `json:"diskEncryptionSet,omitempty"`
+}
+
+// GCPKMSKeyReference gathers required fields for looking up a GCP KMS Key
+type GCPKMSKeyReference struct {
+	// name is the name of the customer-managed encryption key to be used for disk encryption.
+	// The value should correspond to an existing KMS key
+	// and input should match the regular expression: [a-zA-Z0-9_-]{1,63}.
+	// +kubebuilder:validation:Pattern:=^[a-zA-Z0-9_-]$
+	// +kubebuilder:validation:MinLength:=1
+	// +kubebuilder:validation:MaxLength:=63
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// keyRing is the name of the KMS Key Ring which the KMS Key belongs to.
+	// The value should correspond to an existing KMS key ring
+	// and input should match the regular expression: [a-zA-Z0-9_-]{1,63}.
+	// +kubebuilder:validation:Pattern:=^[a-zA-Z0-9_-]$
+	// +kubebuilder:validation:MinLength:=1
+	// +kubebuilder:validation:MaxLength:=63
+	// +kubebuilder:validation:Required
+	KeyRing string `json:"keyRing"`
+
+	// projectID is the ID of the Project in which the KMS Key Ring exists.
+	// It must be 6 to 30 lowercase letters, digits, or hyphens.
+	// It must start with a letter. Trailing hyphens are prohibited.
+	// Defaults to the cluster project defined in the cluster infrastructure object, if not set.
+	// +kubebuilder:validation:Pattern:=^[a-z][a-z0-9-]+[a-z0-9]$
+	// +kubebuilder:validation:MinLength:=6
+	// +kubebuilder:validation:MaxLength:=30
+	// +optional
+	ProjectID string `json:"projectID,omitempty"`
+
+	// location is the GCP location in which the Key Ring exists.
+	// The input must match an existing GCP location, or "global".
```

Comment:
```suggestion
	// The value must match an existing GCP location, or "global".
	// The value should consist of only alphanumeric characters, hyphens (-) and underscores (_).
```
Any length validations on the location?

Reply:
Fixed.